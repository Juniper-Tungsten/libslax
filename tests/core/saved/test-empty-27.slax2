version 1.1;

ns redirect extension = "org.apache.xalan.xslt.extensions.Redirect";

var $test := {
    <x> "ManManMan";
    <x> "Fish Sauce";
    <x> "amazingly awesome sizabilities";
}

match / {
    <out> {
        <in> slax:base64-encode("ManManMan");
        <out> slax:base64-decode("TWFuTWFuTWFu");
        
        for $in ($test/x) {
            var $enc = slax:base64-encode($in);
            var $dec = slax:base64-decode($in);
            
            <test> {
                <in> $in;
                <enc> $enc;
                <dec> $dec;
            }
        }
        <redirect:write href="hello.txt" method="text"> "Hello, World!
";
        <test> slax:document("hello.txt");
        <redirect:write href="hello.txt" method="text"> "TWFuTWFuTWFu";
        var $opts = {
            <format> "base64";
            <encoding> "utf-8";
            <non-xml> "X";
        }
        <test2> slax:document("hello.txt", $opts);
        <redirect:write href="hello.txt" method="text"> "AQwQaAECBXkgc291EAoGAQwMBQ4K";
        <test3> slax:document("hello.txt", $opts);
        var $opts2 = {
            <format> "base64";
            <encoding> "utf-8";
            <non-xml> "á€€";
        }
        <test3> slax:document("hello.txt", $opts2);
    }
}
