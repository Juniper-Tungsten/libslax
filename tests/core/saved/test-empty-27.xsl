<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:redirect="org.apache.xalan.xslt.extensions.Redirect" xmlns:slax-ext="http://xmlsoft.org/XSLT/namespace" xmlns:slax="http://xml.libslax.org/slax" version="1.0" extension-element-prefixes="redirect slax-ext slax">
  <xsl:variable name="test-temp-1">
    <x>ManManMan</x>
    <x>Fish Sauce</x>
    <x>amazingly awesome sizabilities</x>
  </xsl:variable>
  <xsl:variable xmlns:slax-ext="http://xmlsoft.org/XSLT/namespace" name="test" select="slax-ext:node-set($test-temp-1)"/>
  <xsl:template match="/">
    <out>
      <in>
        <xsl:value-of xmlns:slax="http://xml.libslax.org/slax" select="slax:base64-encode(&quot;ManManMan&quot;)"/>
      </in>
      <out>
        <xsl:value-of xmlns:slax="http://xml.libslax.org/slax" select="slax:base64-decode(&quot;TWFuTWFuTWFu&quot;)"/>
      </out>
      <xsl:variable name="slax-dot-1" select="."/>
      <xsl:for-each select="$test/x">
        <xsl:variable name="in" select="."/>
        <xsl:for-each select="$slax-dot-1">
          <xsl:variable xmlns:slax="http://xml.libslax.org/slax" name="enc" select="slax:base64-encode($in)"/>
          <xsl:variable xmlns:slax="http://xml.libslax.org/slax" name="dec" select="slax:base64-decode($in)"/>
          <test>
            <in>
              <xsl:value-of select="$in"/>
            </in>
            <enc>
              <xsl:value-of select="$enc"/>
            </enc>
            <dec>
              <xsl:value-of select="$dec"/>
            </dec>
          </test>
        </xsl:for-each>
      </xsl:for-each>
      <redirect:write href="hello.txt" method="text">
        <xsl:text>Hello, World!
</xsl:text>
      </redirect:write>
      <test>
        <xsl:value-of xmlns:slax="http://xml.libslax.org/slax" select="slax:document(&quot;hello.txt&quot;)"/>
      </test>
      <redirect:write href="hello.txt" method="text">
        <xsl:text>TWFuTWFuTWFu</xsl:text>
      </redirect:write>
      <xsl:variable name="opts">
        <format>base64</format>
        <encoding>utf-8</encoding>
        <non-xml>X</non-xml>
      </xsl:variable>
      <test2>
        <xsl:value-of xmlns:slax="http://xml.libslax.org/slax" select="slax:document(&quot;hello.txt&quot;, $opts)"/>
      </test2>
      <redirect:write href="hello.txt" method="text">
        <xsl:text>AQwQaAECBXkgc291EAoGAQwMBQ4K</xsl:text>
      </redirect:write>
      <test3>
        <xsl:value-of xmlns:slax="http://xml.libslax.org/slax" select="slax:document(&quot;hello.txt&quot;, $opts)"/>
      </test3>
      <xsl:variable name="opts2">
        <format>base64</format>
        <encoding>utf-8</encoding>
        <non-xml>á€€</non-xml>
      </xsl:variable>
      <test3>
        <xsl:value-of xmlns:slax="http://xml.libslax.org/slax" select="slax:document(&quot;hello.txt&quot;, $opts2)"/>
      </test3>
    </out>
  </xsl:template>
</xsl:stylesheet>
