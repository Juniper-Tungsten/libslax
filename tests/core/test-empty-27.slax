version 1.1;

ns redirect extension = "org.apache.xalan.xslt.extensions.Redirect";

var $test := {
    <x> "ManManMan";
    <x> "Fish Sauce";
    <x> "amazingly awesome sizabilities";
}

match / {
    <out> {
	<in> { expr slax:base64-encode("ManManMan"); }
	<out> { expr slax:base64-decode("TWFuTWFuTWFu"); }
	    
	for $in ($test/x) {
	    var $enc = slax:base64-encode($in);
	    var $dec = slax:base64-decode($in);
	    <test> {
		<in> $in;
		<enc> $enc;
		<dec> $dec;
	    }
	}

	<redirect:write href="hello.txt" method="text"> {
	    expr "Hello, World!\n";
	}

	<test> { expr slax:document("hello.txt"); }

	<redirect:write href="hello.txt" method="text"> {
	    expr "TWFuTWFuTWFu";
	}

	var $opts = {
	    <format> "base64";
	    <encoding> "utf-8";
	    <non-xml> "X";
	}

	<test2> { expr slax:document("hello.txt", $opts); }

	<redirect:write href="hello.txt" method="text"> {
	    expr "AQwQaAECBXkgc291EAoGAQwMBQ4K";
	}

	<test3> { expr slax:document("hello.txt", $opts); }

	var $opts3 = {
	    <format> "base64";
	    <encoding> "utf-8";
	    <non-xml> "\u+1000";
	}
	<test3> { expr slax:document("hello.txt", $opts3); }

	var $opts4 = {
	    <format> "base64";
	    <non-xml> "";
	}
	<test4> { expr slax:document("hello.txt", $opts4); }

	var $opts5 = {
	    <format> "base64";
	    <encoding> "utf-8";
	    <non-xml> "[invalid]";
	}
	<test5> { expr slax:document("hello.txt", $opts5); }
    }
}
