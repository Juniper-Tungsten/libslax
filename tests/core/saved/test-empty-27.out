<?xml version="1.0"?>
<out>
  <in>TWFuTWFuTWFu</in>
  <out>ManManMan</out>
  <test>
    <in>ManManMan</in>
    <enc>TWFuTWFuTWFu</enc>
    <dec>ManManMan</dec>
  </test>
  <test>
    <in>Fish Sauce</in>
    <enc>RmlzaCBTYXVjZQ==</enc>
    <dec>Fish Sauce</dec>
  </test>
  <test>
    <in>amazingly awesome sizabilities</in>
    <enc>YW1hemluZ2x5IGF3ZXNvbWUgc2l6YWJpbGl0aWVz</enc>
    <dec>amazingly awesome sizabilities</dec>
  </test>
  <test>Hello, World!
</test>
  <test2>ManManMan</test2>
  <test3>XXXhXXXy souX
XXXXXX
</test3>
  <test3>ကကကhကကကy souက
ကကကကကက
</test3>
</out>
